xgcm/grid_ufunc.py

```diff
@@ -133,7 +147,13 @@ def __repr__(self):
             f"          dask='{self.dask})', map_overlap={self.map_overlap})"
         )
 
-    def __call__(self, grid=None, *args, axis, **kwargs):
+    def __call__(
+        self,
+        grid: "Grid" = None,
```

Comment:
Does `Optional[Grid]` not work here?

Reply:
This should be equivalent.

Reply:
(Although the only reason I made `grid` optional was so that you didn't have to specify it as the first unnamed arg, e.g.

`apply_as_grid_ufunc(da, grid=grid)` rather than `apply_as_grid_ufunc(grid, da)`...)